jaeger/cmd/install.go

```diff
@@ -151,13 +151,14 @@ func render(w io.Writer, valuesOverrides map[string]interface{}, registry string
 	}
 
 	regOrig := vals["webhook"].(map[string]interface{})["image"].(map[string]interface{})["name"].(string)
-	if registry != "" {
-		vals["webhook"].(map[string]interface{})["image"].(map[string]interface{})["name"] = pkgcmd.RegistryOverride(regOrig, registry)
-	}
-	// env var overrides CLI flag
+
 	if override := os.Getenv(flags.EnvOverrideDockerRegistry); override != "" {
 		vals["webhook"].(map[string]interface{})["image"].(map[string]interface{})["name"] = pkgcmd.RegistryOverride(regOrig, override)
 	}
+	// CLI flag overrides env variable
+	if registry != "" {
+		vals["webhook"].(map[string]interface{})["image"].(map[string]interface{})["name"] = pkgcmd.RegistryOverride(regOrig, registry)
```

Comment:
It appears that the env var stopped working for the `linkerd jaeger install` command:

```bash
$ LINKERD_DOCKER_REGISTRY=ENV bin/go-run cli jaeger install --ignore-cluster  | grep image: | sort | uniq
        image: cr.l5d.io/linkerd/jaeger-webhook:dev-09626867-alpeb
        image: jaegertracing/all-in-one:1.31
        image: otel/opentelemetry-collector:0.59.0
```

The reason is `registry` in this line will always be non-empty, holding its default value `pkgcmd.DefaultDockerRegistry`.

As for the previous CI failure, it was just a flake, I re-triggered the tests and they went through :wink:

Reply:
> The reason is registry in this line will always be non-empty, holding its default value

Oh! I assumed that that was what we expected (I donno why I assumed it, makes no sense). I have updated the PR, can you please have a look and let me know if I need to take a different approach? Moreover, I'm kind of new to Go so, pls do let me know if I am not following any convention here.

> As for the previous CI failure, it was just a flake, I re-triggered the tests and they went through 😉

Thanks a lot! I was just amused by the possibility of auto-re-triggers!

Reply:
Thanks for the quick turnaround :+1: 
Can you actually leave the default value for the `--registry` flag as it originally was? IMO those defaults should be static and independent of the environment. Maybe the same approach you had for `multicluster/cmd/link.go` can work here as well? Just using this at the beginning:
```go
	if registry := os.Getenv(flags.EnvOverrideDockerRegistry); registry != "" {
		opts.dockerRegistry = registry
	}
```
and then inside `render()` leave `vals["webhook"].(map[string]interface{})["image"].(map[string]interface{})["name"] = pkgcmd.RegistryOverride(regOrig, registry)` regardless of `registry`'s content...

Reply:
You're right, I take that back, you're doing the same for the link command. And it would require unwarranted added complexity to do it like I suggest.